### PR TITLE
Remove deprecated EnvironmentDependentProviderFactory.isSupported method

### LINKED ISSUE #26280: Remove deprecated EnvironmentDependentProviderFactory.isSupported method
If this is considered a breaking change, we should wait until another major release with the default impl added by #26003 in place (although developers may still not be aware that the method is deprecated, they will be able to proactively resolve the deprecation).

Alternatively if this is not considered a breaking change, then we can immediately remove the old signature.

_Originally posted by @vmuzikar in https://github.com/keycloak/keycloak/issues/26003#issuecomment-1895919188_

### PR DESCRIPTION
**Pull request by pruivo**
Closes #26280

### CODE REVIEW

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for this PR. One more simplification is IMHO possible, see below.

File: server-spi-private/src/main/java/org/keycloak/provider/EnvironmentDependentProviderFactory.java [hunk 1]
```diff
@@ -28,22 +28,12 @@
...
-     * An alternative to {@link #isSupported()} method to check if the provider is supported based on the
-     * provider configuration.
+     * Check if the provider is supported and should be available based on the provider configuration.
      *
      * @param config the provider configuration
      * @return {@code true} if the provider is supported. Otherwise, {@code false}.
      */
     default boolean isSupported(Config.Scope config) {
-        return isSupported();
+        return false;
```
Comment (ahus1) [inline server-spi-private/src/main/java/org/keycloak/provider/EnvironmentDependentProviderFactory.java:37]: IMHO there shouldn't be a default impl here any more, as there is now only a single method in the Interface. Either you implement the interface, or not. 

So please remove the default implementation.

**Review by ahus1 — DISMISSED** [review]
Thank you for this cleanup. This simplified things, and also remove some code areas where the two conditions deviated between the two methods.

**Review by pedroigor — COMMENTED** [review]
I think we want a release notes for this one to let people know that that method was finally removed.

Otherwise, glad we are finally doing this refactoring.

**Review by pedroigor — DISMISSED** [review]
OK, the interface is in `keycloak-server-spi-private` so not a blocker :) But I think it is a good to have.

**Review by ahus1 — APPROVED** [review]

**Comment by ahus1** [comment]
@pedroigor - ok, added it to the release notes and upgrade guide.

**Comment by ahus1** [comment]
@keycloak/ui-maintainers - could you please review this change as the code-owners require you to approve as well? Thanks!

**Review by pedroigor — APPROVED** [review]

**Review by mposolda — APPROVED** [review]
@pruivo Good change, Thanks!

**Review by edewit — APPROVED** [review]

**Pull request merged**